keylime/tpm/tpm_util.py

```diff
@@ -176,3 +179,140 @@ def checkquote(
         raise Exception("The digest of the quoteblob differs from the quote's digest")
 
     return pcrs_dict
+
+
+def makecredential(ek_tpm: bytes, challenge: bytes, aik_name: bytes) -> bytes:
+    public_key, hash_alg, _, _ = tpm2_objects.pubkey_parms_from_tpm2b_public(ek_tpm)
+    # python crypto only supports encryption with RSA public key
+    if not isinstance(public_key, (RSAPublicKey)):
```

Comment:
Removed...

Reply:
FYI: Due to python crypto we cannot easily use EC EK keys since makecredential doesn't work for EC keys. For EC key support we would probably have to resort to openssl command line tool or another crypto library...

Reply:
> FYI: Due to python crypto we cannot easily use EC EK keys since makecredential doesn't work for EC keys. For EC key support we would probably have to resort to openssl command line tool or another crypto library...

What is the issue with python crypto and EC keys?

Reply:
> > FYI: Due to python crypto we cannot easily use EC EK keys since makecredential doesn't work for EC keys. For EC key support we would probably have to resort to openssl command line tool or another crypto library...
> 
> What is the issue with python crypto and EC keys?

```# python crypto only supports encryption with RSA public key```

Reply:
@ansasaki Encryption with EC keys does exist: https://github.com/stefanberger/libtpms/blob/master/src/tpm2/crypto/openssl/CryptEccCrypt.c#L101   

> If makecredential is dependent on encryption using an RSA key, then there is no way to workaround this by using openssl.

When EC keys are used we would need to implement something like this here, which seems to be a different method: https://github.com/stefanberger/libtpms/blob/master/src/tpm2/CryptUtil.c#L504-L558

Reply:
If the EK is an EC key and to be used for MakeCredential then the biggest concerns for implementing CryptSecretEncrypt are the functions 
1) CryptEccIsPointOnCurve and 
2) CryptEccPointMultiply. 

For 1) this here may work: https://github.com/pyca/cryptography/blob/main/src/cryptography/hazmat/primitives/asymmetric/ec.py#L169

For 2) We may have to do something like this here: **https://github.com/pyca/cryptography/blob/c7146f9ed3830a9c3120722bc99dda81ce77037d/src/cryptography/hazmat/backends/openssl/backend.py#L1228-L1230**